src/boot/boot.c

```diff
@@ -1279,7 +1299,9 @@ static void config_defaults_load_from_file(Config *config, char *content) {
                                           value);
 
                 } else if (streq8(key, "reboot-on-error")) {
-                        if (!parse_boolean(value, &config->reboot_on_err))
+                        if (streq8(value, "auto"))
+                                config->reboot_on_err = REBOOT_AUTO;
+                        else if (!parse_boolean(value, (bool *) &config->reboot_on_err))
```

Comment:
pleas don't do that cast. yes, in practice bool and the enum will both be implemented as int. but that's not universally the case. just write it to a temporarily bool variable please, and assign it to reboot_on_error then